app/test_main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+import pytest
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, valid_url_return, internet_return, expected_result",
+    [
+        ("https://www.google.com", True, True, "Accessible"),
+        ("https://www.full-sheet-url.co", False, True, "Not accessible"),
+        ("https://www.google.com", True, False, "Not accessible"),
+        ("https://www.full-sheet-url.co", False, False, "Not accessible"),
+    ]
+)
+@patch("app.main.valid_google_url")
+@patch("app.main.has_internet_connection")
+def test_can_access_google_page(
```

Comment:
The test function name `test_can_access_google_page` could be more descriptive. Consider renaming it to something like `test_accessibility_based_on_url_and_connection` to better reflect the scenarios being tested, as suggested in the checklist: 'Use descriptive and correct tests names.'